08_Interactive_Image_Visualization.ipynb

```diff
@@ -0,0 +1,1274 @@
+   "source": [
+    "> STOP - Select some data points from the plot above using the box select tool before proceeding"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "selection = (points.data.x > box.bounds[0]) \\\n",
+    "    & (points.data.y > box.bounds[1]) \\\n",
+    "    & (points.data.x < box.bounds[2]) \\\n",
+    "    & (points.data.y < box.bounds[3])\n",
+    "print('The selection box contains %i datapoints'%(np.sum(selection)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will plot a spatial distribution on the sky of all the data and link it to a two-dimensional histogram of the data in the box selection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First, create a holoviews dataset instance. \n",
+    "# Here we label some of the columns.\n",
+    "kdims = [('ra', 'RA(deg)'), ('dec', 'Dec(deg)')]\n",
+    "vdims = [('mag_r_cModel', 'r(mag)')]\n",
+    "ds = hv.Dataset(source.to_df(), kdims, vdims)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "points = hv.Points(ds)\n",
+    "boundsxy = (np.min(ds.data['ra']), np.min(ds.data['dec']),\n",
+    "            np.max(ds.data['ra']), np.max(ds.data['dec']))\n",
+    "box = streams.BoundsXY(source=points, bounds=boundsxy)\n",
+    "box_plot = hv.DynamicMap(lambda bounds: hv.Bounds(bounds), streams=[box])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Custom callback functionality to update the linked histogram\n",
+    "def log_inf(x):\n",
+    "    return np.log(x) if x > 0 else 0\n",
+    "\n",
+    "\n",
+    "def update_histogram(bounds=bounds):\n",
+    "    selection = (ds.data['ra'] > bounds[0]) & \\\n",
+    "                (ds.data['dec'] > bounds[1]) & \\\n",
+    "                (ds.data['ra'] < bounds[2]) & \\\n",
+    "                (ds.data['dec'] < bounds[3])\n",
+    "\n",
+    "    selected_mag = ds.data.loc[selection]['mag_r_cModel']\n",
+    "    frequencies, edges = np.histogram(selected_mag)\n",
+    "    hist = hv.Histogram((list(map(log_inf, frequencies)), edges))\n",
+    "    return hist"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dmap = hv.DynamicMap(update_histogram, streams=[box]).options(\n",
+    "    height=400, width=400)\n",
+    "datashade(points,\n",
+    "          cmap=process_cmap(\"Viridis\", provider=\"bokeh\")) * \\\n",
+    "box_plot.options(height=400, width=400) + \\\n",
+    "dmap"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Try changing the box selection and watch as the histogram is recomputed and displayed. "
```

Comment:
The histogram is not being redrawn for me (it doesn't update with my box selection).

Reply:
Melissa repoorted the same issue in the slack channel

Reply:
This does work when I re-ran the entire notebook (one "run all"). Not sure if it could have been some focus change or something when I was going through before.

Reply:
Excellent!